### PR TITLE
598 preprocessor script for printing the library versions to boot outtxt

### PR DESCRIPTION
**Pull request by tyeth**

### CODE REVIEW

**Comment by brentru** [comment]
@tyeth  Is this ready for review?

**Comment by tyeth** [comment]
This **should not be merged** until the ci-wippersnapper branch for ci-arduino has been updated, as part of  https://github.com/adafruit/ci-arduino/pull/196, and this PR had the refs updated in the build script. 
I probably need to resolve the conflict with the build file anyway.
Alternatively the current (598-blah) ref will work as it's ci-wippersnapper + print dependencies changes.

**Comment by tyeth** [comment]
@brentru this should be ready for review

File: .github/workflows/build-clang-doxy.yml [hunk 1]
```diff
@@ -578,9 +582,9 @@ jobs:
           echo >>$GITHUB_ENV WS_VERSION=$(git describe --dirty --tags)
       - uses: actions/checkout@v4
         with:
-          repository: adafruit/ci-arduino
+          repository: tyeth/ci-arduino
```
Comment (brentru) [inline .github/workflows/build-clang-doxy.yml:585]: Would we need to modify the ci-arduino `wippersnapper` branch to match this `ref`?

File: .github/workflows/build-clang-doxy.yml [hunk 2]
```diff
@@ -553,7 +557,7 @@ jobs:
             wippersnapper.${{ matrix.arduino-platform }}.littlefs.${{ env.WS_VERSION }}.zip
 
   build-esp32sx-dev:
-    name: Build WipperSnapper ESP32-Sx DEV BUILDS
+    name: 🏗️ESP32-Sx DEV BUILDS
```
Comment (brentru) [inline .github/workflows/build-clang-doxy.yml:560]: Are we only doing this for Sx builds or all builds?

File: .github/workflows/build-clang-doxy.yml [hunk 3]
```diff
@@ -610,7 +614,7 @@ jobs:
         run: |
           cp /home/runner/Arduino/libraries/Adafruit_LittlevGL_Glue_Library/lv_conf.h /home/runner/Arduino/libraries
       - name: Build for ESP32-SX
-        run: python3 ci/build_platform.py ${{ matrix.arduino-platform }} --build_timeout 48000
+        run: python3 ci/build_platform.py ${{ matrix.arduino-platform }} --build_timeout 48000 --include_print_dependencies_header ./src/print_dependencies.h
```
Comment (brentru) [inline .github/workflows/build-clang-doxy.yml:617]: Should we do this on all platforms, or just ESP32-x?

File: src/print_dependencies.cpp [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#include "print_dependencies.h"
+
+const char *project_dependencies = R"(
```
Comment (brentru) [inline src/print_dependencies.cpp:3]: How does this work? As I understand from reading thru everything, it populates `project_dependencies` with a list of dependencies by using `arduino-cli lib list` during pre-compilation. Or is this during compilation?

File: src/provisioning/littlefs/WipperSnapper_LittleFS.cpp [hunk 5]
```diff
@@ -28,6 +28,14 @@
 */
 /**************************************************************************/
 WipperSnapper_LittleFS::WipperSnapper_LittleFS() {
+#if PRINT_DEPENDENCIES
+  // Print project build dependencies
+  WS_DEBUG_PRINTLN("Build Dependencies:");
+  WS_DEBUG_PRINTLN("*********************");
+  WS_DEBUG_PRINTLN(project_dependencies);
```
Comment (brentru) [inline src/provisioning/littlefs/WipperSnapper_LittleFS.cpp:35]: Nice, really simple way to print out the ptr to the list

**Review by brentru — CHANGES_REQUESTED** [review]
@tyeth  Thanks for taking this on, its a simple approach. I do have some q's

**Pull request merged**